### PR TITLE
Add a precommit hook that runs linting

### PR DESCRIPTION
**Pull request by talldan**
Uses husky to add a precommit hook. The hook currently lints:
- js
- css
- package.json

It fails when there are linting errors in a staged file.

Also adds a .editorconfig (stolen from Gutenberg).

### CODE REVIEW

File: .lintstagedrc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (pento) [inline .lintstagedrc]: Could these be in package.json, same as Gutenberg does?

File: .lintstagedrc [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "packages/*/package.json": [
```
Comment (pento) [inline .lintstagedrc]: There is no `packages` directory. This could just run on `package.json`, I guess?

File: package.json [hunk 3]
```diff
@@ -49,6 +49,8 @@
 		"cross-env": "5.2.0",
 		"electron": "4.0.5",
 		"electron-builder": "20.38.5",
+		"husky": "^1.3.1",
```
Comment (pento) [inline package.json]: Need to use exact versions, rather than matching all minor releases.

File: .lintstagedrc (hunk 1 above)
Comment (talldan) [reply .lintstagedrc]: 👍

File: .lintstagedrc (hunk 2 above)
Comment (talldan) [reply .lintstagedrc]: good catch 👍

File: package.json (hunk 3 above)
Comment (talldan) [reply package.json]: Done

**Pull request merged**